pymc3/ode/ode.py

```diff
@@ -1,211 +1,222 @@
+
+        If the last n_theta entries of the parameters vector in the simulate call
+        correspond to ode paramaters, then the last n_theta columns in
+        the sensitivity matrix will be 0.        
+        """
+
+        # Initialize the sensitivity matrix to be 0 everywhere
+        sens_matrix = np.zeros((self.n_states, self.n_states + self.n_theta), dtype=floatX)
+
+        # Slip in the identity matrix in the appropirate place
+        sens_matrix[:,:self.n_states] = np.eye(self.n_states, dtype=floatX)
+
+        # We need the sensitivity matrix to be a vector (see augmented_function)
+        # Ravel and return
+        dydp = sens_matrix.ravel()
+        return dydp
+
+    def _system(self, Y, t, p):
+        """This is the function that will be passed to odeint. Solves both ODE and sensitivities.
+
+        Args:
+            Y: augmented state vector (n_states + n_states + n_theta)
+            t: current time
+            p: parameter vector (y0, theta)
+        """
+        dydt, ddt_dydp = self._augmented_func(Y[:self.n_states], t, p, Y[self.n_states:])
+        derivatives = np.concatenate([dydt, ddt_dydp])
+        return derivatives
+
+    def _simulate(self, y0, theta):
+        # Initial condition comprised of state initial conditions and raveled sensitivity matrix
+        s0 = np.concatenate([y0, self._sens_ic])
+
+        # perform the integration
+        sol = scipy.integrate.odeint(
+            func=self._system, y0=s0, t=self._augmented_times, args=(np.concatenate([y0, theta]),)
+        ).astype(floatX)
+        # The solution
+        y = sol[1:, :self.n_states]
+
+        # The sensitivities, reshaped to be a sequence of matrices
+        sens = sol[1:, self.n_states:].reshape(self.n_times, self.n_states, self.n_p)
+
+        return y, sens
+
+    def make_node(self, y0, theta):
+        inputs = (y0, theta)
+        _log.debug('make_node for inputs {}'.format(hash(inputs)))
+        states = self._otypes[0]()
+        sens = self._otypes[1]()
+
+        # store symbolic output in dictionary such that it can be accessed in the grad method
+        self._output_sensitivities[hash(inputs)] = sens
+        return theano.Apply(self, inputs, (states, sens))
+
+    def __call__(self, y0, theta, return_sens=False, **kwargs):
+        if isinstance(y0, (list, tuple)) and not len(y0) == self.n_states:
+            raise ShapeError('Length of y0 is wrong.', actual=(len(y0),), expected=(self.n_states,))
+        if isinstance(theta, (list, tuple)) and not len(theta) == self.n_theta:
+            raise ShapeError('Length of theta is wrong.', actual=(len(theta),), expected=(self.n_theta,))
+            
+        # convert inputs to tensors (and check their types)
+        y0 = tt.cast(tt.unbroadcast(tt.as_tensor_variable(y0), 0), floatX)
+        theta = tt.cast(tt.unbroadcast(tt.as_tensor_variable(theta), 0), floatX)
+        inputs = [y0, theta]
+        for i, (input, itype) in enumerate(zip(inputs, self._itypes)):
```

Comment:
`input` is unfortunately a builtin (for prompting user input). We should never use it as a builtin, but changing it would be good.

Reply:
Good point. I should do a follow-up PR to rename it